File: expresslk-services/ExpressLK-core/src/main/java/lk/express/bean/BusLight.java
package lk.express.bean;

import java.math.BigInteger;
import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.xml.bind.annotation.XmlRootElement;
import javax.xml.bind.annotation.XmlType;

@Entity
@Table(name = "bus_light")
@XmlType(name = "BusLight", namespace = "http://bean.express.lk")
@XmlRootElement
public class BusLight extends LightEntity implements HasAllowedDivisions {

	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue
	private Integer id;
	@Column(name = "plate_number")
	private String plateNumber;
	@Column(name = "name")
	private String name;
	@Column(name = "contact")
	private String contact;
	@Column(name = "admin_contact")
	private String adminContact;
	@Column(name = "bus_type_id")
	private Integer busType;
	@Column(name = "travel_class_id")
	private Integer travelClass;
	@Column(name = "supplier_id")
	private Integer supplier;
	@Column(name = "division_id")
	private Integer division;
	@Column(name = "notification_method")
	private String notificationMethod;
	@Column(name = "rating")
	private Float rating;

	@Column(name = "markup_only", columnDefinition = "bit")
	@org.hibernate.annotations.Type(type = "org.hibernate.type.NumericBooleanType")
	private boolean markupOnly;
	@Column(name = "cash_on_departure_allowed", columnDefinition = "bit")
	@org.hibernate.annotations.Type(type = "org.hibernate.type.NumericBooleanType")
	private boolean cashOnDepartureAllowed;

	@Column(name = "allowed_divisions")
	private BigInteger allowedDivisions;

	@Column(name = "driver_id")
	private Integer driver;
	@Column(name = "conductor_id")
	private Integer conductor;
	@Column(name = "seating_profile_id")
	private Integer seatingProfile;

	@Column(name = "seconds_before_tb_end")
	private Integer secondsBeforeTBEnd;
	@Column(name = "seconds_before_web_end")
	private Integer secondsBeforeWebEnd;
	@Column(name = "seconds_before_ticketing_active")
	private Integer secondsBeforeTicketingActive;

	@Column(name = "permit_number")
	private String permitNumber;
	@Temporal(TemporalType.DATE)
	@Column(name = "permit_issue_date")
	private Date permitIssueDate;
	@Temporal(TemporalType.DATE)
	@Column(name = "permit_expiry_date")
	private Date permitExpiryDate;

	@Override
	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getPlateNumber() {
		return plateNumber;
	}

	public void setPlateNumber(String plateNumber) {
		this.plateNumber = plateNumber;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getContact() {
		return contact;
	}

	public void setContact(String contact) {
		this.contact = contact;
	}

	public String getAdminContact() {
		return adminContact;
	}

	public void setAdminContact(String adminContact) {
		this.adminContact = adminContact;
	}

	public Integer getBusType() {
		return busType;
	}

	public void setBusType(Integer busType) {
		this.busType = busType;
	}

	public Integer getTravelClass() {
		return travelClass;
	}

	public void setTravelClass(Integer travelClass) {
		this.travelClass = travelClass;
	}

	public Integer getSupplier() {
		return supplier;
	}

	public void setSupplier(Integer supplier) {
		this.supplier = supplier;
	}

	public Integer getDivision() {
		return division;
	}

	public void setDivision(Integer division) {
		this.division = division;
	}

	public String getNotificationMethod() {
		return notificationMethod;
	}

	public void setNotificationMethod(String notificationMethod) {
		this.notificationMethod = notificationMethod;
	}

	public Float getRating() {
		return rating;
	}

	public void setRating(Float rating) {
		this.rating = rating;
	}

	public boolean isMarkupOnly() {
		return markupOnly;
	}

	public void setMarkupOnly(boolean markupOnly) {
		this.markupOnly = markupOnly;
	}

	public boolean isCashOnDepartureAllowed() {
		return cashOnDepartureAllowed;
	}

	public void setCashOnDepartureAllowed(boolean cashOnDepartureAllowed) {
		this.cashOnDepartureAllowed = cashOnDepartureAllowed;
	}

	@Override
	public BigInteger getAllowedDivisions() {
		return allowedDivisions;
	}

	@Override
	public void setAllowedDivisions(BigInteger allowedDivisions) {
		this.allowedDivisions = allowedDivisions;
	}

	public Integer getDriver() {
		return driver;
	}

	public void setDriver(Integer driver) {
		this.driver = driver;
	}

	public Integer getConductor() {
		return conductor;
	}

	public void setConductor(Integer conductor) {
		this.conductor = conductor;
	}

	public Integer getSeatingProfile() {
		return seatingProfile;
	}

	public void setSeatingProfile(Integer seatingProfile) {
		this.seatingProfile = seatingProfile;
	}

	public Integer getSecondsBeforeTBEnd() {
		return secondsBeforeTBEnd;
	}

	public void setSecondsBeforeTBEnd(Integer secondsBeforeTBEnd) {
		this.secondsBeforeTBEnd = secondsBeforeTBEnd;
	}

	public Integer getSecondsBeforeWebEnd() {
		return secondsBeforeWebEnd;
	}

	public void setSecondsBeforeWebEnd(Integer secondsBeforeWebEnd) {
		this.secondsBeforeWebEnd = secondsBeforeWebEnd;
	}

	public Integer getSecondsBeforeTicketingActive() {
		return secondsBeforeTicketingActive;
	}

	public void setSecondsBeforeTicketingActive(Integer secondsBeforeTicketingActive) {
		this.secondsBeforeTicketingActive = secondsBeforeTicketingActive;
	}

	public String getPermitNumber() {
		return permitNumber;
	}

	public void setPermitNumber(String permitNumber) {
		this.permitNumber = permitNumber;
	}

	public Date getPermitIssueDate() {
		return permitIssueDate;
	}

	public void setPermitIssueDate(Date permitIssueDate) {
		this.permitIssueDate = permitIssueDate;
	}

	public Date getPermitExpiryDate() {
		return permitExpiryDate;
	}

	public void setPermitExpiryDate(Date permitExpiryDate) {
		this.permitExpiryDate = permitExpiryDate;
	}

}

File: expresslk-webportal/module/Api/src/Api/Client/Soap/Core/BusLight.php
<?php

namespace Api\Client\Soap\Core;

include_once('LightEntity.php');

class BusLight extends LightEntity
{

    /**
     * @var string $adminContact
     * @access public
     */
    public $adminContact = null;

    /**
     * @var int $allowedDivisions
     * @access public
     */
    public $allowedDivisions = null;

    /**
     * @var int $busType
     * @access public
     */
    public $busType = null;

    /**
     * @var boolean $cashOnDepartureAllowed
     * @access public
     */
    public $cashOnDepartureAllowed = null;

    /**
     * @var int $conductor
     * @access public
     */
    public $conductor = null;

    /**
     * @var string $contact
     * @access public
     */
    public $contact = null;

    /**
     * @var int $division
     * @access public
     */
    public $division = null;

    /**
     * @var int $driver
     * @access public
     */
    public $driver = null;

    /**
     * @var int $id
     * @access public
     */
    public $id = null;

    /**
     * @var boolean $markupOnly
     * @access public
     */
    public $markupOnly = null;

    /**
     * @var string $name
     * @access public
     */
    public $name = null;

    /**
     * @var string $notificationMethod
     * @access public
     */
    public $notificationMethod = null;

    /**
     * @var dateTime $permitExpiryDate
     * @access public
     */
    public $permitExpiryDate = null;

    /**
     * @var dateTime $permitIssueDate
     * @access public
     */
    public $permitIssueDate = null;

    /**
     * @var string $permitNumber
     * @access public
     */
    public $permitNumber = null;

    /**
     * @var string $plateNumber
     * @access public
     */
    public $plateNumber = null;

    /**
     * @var float $rating
     * @access public
     */
    public $rating = null;

    /**
     * @var int $seatingProfile
     * @access public
     */
    public $seatingProfile = null;

    /**
     * @var int $secondsBeforeTBEnd
     * @access public
     */
    public $secondsBeforeTBEnd = null;

    /**
     * @var int $secondsBeforeTicketingActive
     * @access public
     */
    public $secondsBeforeTicketingActive = null;

    /**
     * @var int $secondsBeforeWebEnd
     * @access public
     */
    public $secondsBeforeWebEnd = null;

    /**
     * @var int $supplier
     * @access public
     */
    public $supplier = null;

    /**
     * @var int $travelClass
     * @access public
     */
    public $travelClass = null;

    /**
     * @param string $adminContact
     * @param int $allowedDivisions
     * @param int $busType
     * @param boolean $cashOnDepartureAllowed
     * @param int $conductor
     * @param string $contact
     * @param int $division
     * @param int $driver
     * @param int $id
     * @param boolean $markupOnly
     * @param string $name
     * @param string $notificationMethod
     * @param dateTime $permitExpiryDate
     * @param dateTime $permitIssueDate
     * @param string $permitNumber
     * @param string $plateNumber
     * @param float $rating
     * @param int $seatingProfile
     * @param int $secondsBeforeTBEnd
     * @param int $secondsBeforeTicketingActive
     * @param int $secondsBeforeWebEnd
     * @param int $supplier
     * @param int $travelClass
     * @access public
     */
    public function __construct($adminContact, $allowedDivisions, $busType, $cashOnDepartureAllowed, $conductor, $contact, $division, $driver, $id, $markupOnly, $name, $notificationMethod, $permitExpiryDate, $permitIssueDate, $permitNumber, $plateNumber, $rating, $seatingProfile, $secondsBeforeTBEnd, $secondsBeforeTicketingActive, $secondsBeforeWebEnd, $supplier, $travelClass)
    {
      $this->adminContact = $adminContact;
      $this->allowedDivisions = $allowedDivisions;
      $this->busType = $busType;
      $this->cashOnDepartureAllowed = $cashOnDepartureAllowed;
      $this->conductor = $conductor;
      $this->contact = $contact;
      $this->division = $division;
      $this->driver = $driver;
      $this->id = $id;
      $this->markupOnly = $markupOnly;
      $this->name = $name;
      $this->notificationMethod = $notificationMethod;
      $this->permitExpiryDate = $permitExpiryDate;
      $this->permitIssueDate = $permitIssueDate;
      $this->permitNumber = $permitNumber;
      $this->plateNumber = $plateNumber;
      $this->rating = $rating;
      $this->seatingProfile = $seatingProfile;
      $this->secondsBeforeTBEnd = $secondsBeforeTBEnd;
      $this->secondsBeforeTicketingActive = $secondsBeforeTicketingActive;
      $this->secondsBeforeWebEnd = $secondsBeforeWebEnd;
      $this->supplier = $supplier;
      $this->travelClass = $travelClass;
    }

}
